src/graph.js

```diff
@@ -40,37 +40,135 @@ class Graph {
   // Optionally accepts an array of other GraphNodes for the new vertex to be connected to
   // Returns the newly-added vertex
   addVertex(value, edges = []) {
-
+    const newGN = new GraphNode({ value, edges });
+    
+    if (this.vertices.length === 0) {
+      this.vertices.push(newGN);
+    } else if (this.vertices.length === 1) {
+      this.vertices.push(newGN);
+      this.vertices[0].pushToEdges(newGN);
+      newGN.pushToEdges(this.vertices[0]);
+    } else {
+      this.vertices.push(newGN);
+      newGN.edges.forEach((node) => {
+        node.pushToEdges(newGN);
+      });
+      // newGN.edges().forEach((ngeNode) => {
+      //   this.vertices.forEach((vNode) => {
+      //     if (ngeNode.value() === vNode.value()) {
+      //       vNode.pushToEdges(newGN);
+      //     }
+      //   });
+      // });
+      // this.vertices.forEach((vert) => {
+      //   newGN.edges().forEach((newGNConnection) => {
+      //     if (newGNConnection === vert.value()) {
+      //     vert.pushToEdges(newGN); 
+      //     }
+      //   });
+      //   vert.edges().forEach((connectionNode) => {
+      //     if (connectionNode.value() === newGN.value() && The connection node doesn't already exist) {
+      //       newGN.pushToEdges(connectionNode);
+      //     }
+      //   })
+      // });
+    }
+    return newGN;
   }
   // Checks all the vertices of the graph for the target value
   // Returns true or false
   contains(value) {
-
+    let found = false;
+    this.vertices.forEach((node) => {
+      if (node.value === value) {
+        found = true;
+      }
+    });
+    return found;
   }
   // Checks the graph to see if a GraphNode with the specified value exists in the graph 
   // and removes the vertex if it is found
   removeVertex(value) {
-
+    for (let i = 0; i < this.vertices.length; i++) {
+      if (this.vertices[i].value === value) {
+        this.vertices.splice(i, 1);  
+      }
+    }
   }
   // Checks the two input vertices to see if each one references the other in their respective edges array
   // Both vertices must reference each other for the edge to be considered valid
   // If only one vertex references the other but not vice versa, should not return true
   // Note: You'll need to store references to each vertex's array of edges so that you can use 
   // array methods on said arrays. There is no method to traverse the edge arrays built into the GraphNode class
   checkIfEdgeExists(fromVertex, toVertex) {
-
+    const fromEdges = fromVertex.edges;
+    const toEdges = toVertex.edges;
+    
+    let checkEdge = false;
+    fromEdges.forEach((efNode) => {
```

Comment:
I would try to avoid nested loops here since it makes your implementation run in quadratic (O(n^2)) time instead of linear time (O(n)).